File: src/main/java/com/ptaf/ai/model/GenerationResult.java
package com.ptaf.ai.model;

import com.ptaf.ai.validation.StepReuseValidationResult;
import com.ptaf.ai.validation.YamlKeyValidationResult;
import com.ptaf.ai.validation.AllowedYamlGuardResult;
import com.ptaf.ai.validation.MissingYamlPatchSuggestion;
import com.ptaf.ai.validation.PageFrameContextGuardResult;
import com.ptaf.ai.validation.RunnableFeatureResult;

import java.util.List;

/**
 * Model output plus Phase 2 local reuse trace (keyword-ranked patterns with source paths).
 */
public record GenerationResult(
        String featureGherkin,
        List<String> suggestedReusableSteps,
        String rawModelResponse,
        List<ScoredPattern> reuseTrace,
        AiGenerationStructuredResponse structuredResponse,
        StepReuseValidationResult stepReuseValidationResult,
        YamlKeyValidationResult yamlKeyValidationResult,
        AllowedYamlGuardResult allowedYamlGuardResult,
        PageFrameContextGuardResult pageFrameContextGuardResult,
        RunnableFeatureResult runnableFeatureResult,
        List<MissingYamlPatchSuggestion> missingYamlPatchSuggestions
) {
}

File: python/ptaf/ai/model/generation_result.py
from __future__ import annotations

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ptaf.ai.model.ai_generation_structured_response import AiGenerationStructuredResponse
    from ptaf.ai.validation.allowed_yaml_guard_result import AllowedYamlGuardResult
    from ptaf.ai.validation.missing_yaml_patch_suggestion import MissingYamlPatchSuggestion
    from ptaf.ai.validation.page_frame_context_guard_result import PageFrameContextGuardResult
    from ptaf.ai.validation.runnable_feature_result import RunnableFeatureResult
    from ptaf.ai.validation.step_reuse_validation_result import StepReuseValidationResult
    from ptaf.ai.validation.yaml_key_validation_result import YamlKeyValidationResult


@dataclass
class GenerationResult:
    feature_gherkin: str
    suggested_reusable_steps: list[str]
    raw_model_response: str
    reuse_trace: list
    structured_response: AiGenerationStructuredResponse
    step_reuse_validation_result: StepReuseValidationResult | None
    yaml_key_validation_result: YamlKeyValidationResult | None
    allowed_yaml_guard_result: AllowedYamlGuardResult | None
    page_frame_context_guard_result: PageFrameContextGuardResult | None
    runnable_feature_result: RunnableFeatureResult | None
    missing_yaml_patch_suggestions: list[MissingYamlPatchSuggestion]
